hardware/raspberrypi/booteeprom.md

```diff
@@ -7,38 +7,97 @@ The Raspberry Pi 4 has an SPI-attached EEPROM (4MBits/512KB), which contains cod
+
+```
+# Prevent the service from running, this can be run before the package is installed to prevent it ever running automatically.
+sudo systemctl mask rpi-eeprom-update
+
+# Enable it again
+sudo systemctl unmask rpi-eeprom-update
+```
+
+The *FREEZE_VERSION* configuration option the EEPROM configuration may be used to indicate that the EEPROM should not be updated on this board. 
+
 ## Write protection of EEPROM
 
 There is no software write protection for the boot EEPROM but there will be a mechanism in Raspbian to skip any future updates to the EEPROM. However, it is possible to physically write-protect both EEPROMs via a simple resistor change on the board. Details will be published in the [schematics](./schematics/README.md).
 
 ## EEPROM configuration options
 
-The EEPROM image contains a small user-modifiable config file. To change a setting:
+The EEPROM image contains a small user-modifiable config file, the rpi-eeprom package contains a tool called rpi-eeprom-config to read and modify the bootloader config file in an EEPROM image (pieeprom.bin) file.
+
+### Update the EEPROM config
+```
+# Copy the EEPROM of interest from /lib/firmware/raspberrypi/bootloader/critical/
+
+# To extract the configuration file from an EEPROM image.
+rpi-eeprom-config pieeprom.bin --out bootconf.txt
+
+# To update the configuration file in an EEPROM image.
+rpi-eeprom-config pieeprom.bin --config bootconf.txt --out pieeprom-new.bin
+
+# To flash the new image
+# -d means that the configuration in the file should be used, otherwise, rpi-eeprom-update 
+# will automatically migrate the current bootloader's configuration to the new image.
+sudo rpi-eeprom-update -d -f ./pieeprom-new.bin
+```
+
+### Checking if an update is available
+Running the rpi-eeprom-update command with no parameters indicates whether an update is required. An update is required if the timestamp of the most recent file in the firmware directory (normally /lib/firmware/raspberrypi/bootloader/critical) is newer than that reported
+by the current bootloader.
+```
+rpi-eeprom-update
+```
+
+### Reading the current EEPROM configuration
+
+To view the configuration file used by the bootloader at boot type
+```
+vcgencmd bootloader_config
+```
+
+### Reading the EEPROM version
+```
+vcgencmd bootloader_version
+```
 
-* Download and unzip the rescue bootloader image from https://www.raspberrypi.org/downloads/
-* Used sed to change setting. Be careful to avoid changing anything else otherwise it will fail to boot.
-  * `sed -i -e "s/BOOT_UART=0/BOOT_UART=1/" pieeprom.bin`
-* Flash the image - see embedded README.txt in rescue image zip.
+### Beta firmware
+Beta firmware files will be stored in /lib/firmware/raspberrypi/bootloader/critical/. Developers or beta-testers who are comfortable with using the rescue image to fix boot problems can track the beta firmware by editing /etc/default/rpi-eeprom-update 
+```
+Change FIRMWARE_RELEASE_STATUS="critical"
+to FIRMWARE_RELEASE_STATUS="beta"
```

Comment:
Are these beta files still stored in `/lib/firmware/raspberrypi/bootloader/critical/`, or was that path supposed to be `/lib/firmware/raspberrypi/bootloader/beta/` ?